src/client/mod.rs

```diff
@@ -90,31 +95,52 @@ impl GaloyClient {
         Ok(me)
     }
 
-    pub fn request_auth_code(&self, phone: String) -> anyhow::Result<bool> {
-        let input = UserRequestAuthCodeInput { phone };
-
-        let variables = user_request_auth_code::Variables { input };
-
-        let response_body =
-            post_graphql::<UserRequestAuthCode, _>(&self.graphql_client, &self.api, variables)
-                .context("issue fetching response")?;
-
-        let response_data = response_body.data.context("Query failed or is empty")?; // TODO: understand when this can fail here
-
-        let UserRequestAuthCodeUserRequestAuthCode { success, errors } =
-            response_data.user_request_auth_code;
-
-        match success {
-            Some(true) => Ok(true),
-            _ if !errors.is_empty() => {
-                println!("{:?}", errors);
-                bail!("request failed (graphql errors)")
+    pub fn request_phone_code(&self, phone: String, nocaptcha: bool) -> std::io::Result<()> {
+        match nocaptcha {
+            false => {
+                let listener = TcpListener::bind("127.0.0.1:0")?;
+                let port = listener.local_addr().unwrap().port();
```

Comment:
`unwrap` not good to have in production but this can be handled post-merge in the PR I have to improve error handling across board